src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -3,34 +3,72 @@
+    Rectangle inflated_defense_area =
+        buildInflatedDefenseArea(field, robot_obstacle_inflation_factor);
+    // Look directly at the point
+    Vector block_vec     = (field.friendlyGoalCenter() - enemy_threat_origin);
+    Angle angle_to_block = block_vec.orientation();
+    // Shot ray to block
+    Ray ray(enemy_threat_origin, angle_to_block);
+    std::optional<Point> block_point =
+        findDefenseAreaIntersection(field, ray, inflated_defense_area);
+    if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::LEFT &&
+        block_point.has_value())
     {
-        angle_to_block = angle_to_positive_goalpost + shot_angle_sixth * 1.0;
+        // rotate pos 90 deg
+        block_point = stepAlongPerimeter(inflated_defense_area, block_point.value(),
+                                         -ROBOT_MAX_RADIUS_METERS * 2);
+        // block_point never goes out of bounds because inflated defense area is a
+        // rectangle that is always within the field. However, it may be out of "playable"
+        // boundaries, which isn't useful to us so we move it back inside.
+        if (!contains(field.fieldLines(), block_point.value()))
+        {
+            /*  In this scenario 0 moves towards x to come back in play.
+             *            +-------
+             *          O |
+             *       X    |
+             * -------\ O +------
+             *         \
+             *          0
+             */
+            block_point =
+                block_point.value() + Vector(ROBOT_MAX_RADIUS_METERS * 2 + x_shift_meters,
+                                             ROBOT_MAX_RADIUS_METERS * 2);
+        }
     }
-    else if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::RIGHT)
+    else if (crease_defender_alignment == TbotsProto::CreaseDefenderAlignment::RIGHT &&
+             block_point.has_value())
     {
-        angle_to_block = angle_to_positive_goalpost + shot_angle_sixth * 5.0;
+        // rotate neg 90 deg
+        block_point = stepAlongPerimeter(inflated_defense_area, block_point.value(),
+                                         ROBOT_MAX_RADIUS_METERS * 2);
+        // block_point never goes out of bounds because inflated defense area is a
+        // rectangle that is always within the field. However, it may be out of "playable"
+        // boundaries, which isn't useful to us so we move it back inside.
+        if (!contains(field.fieldLines(), block_point.value()))
+        {
+            /*  In this scenario 0 moves towards x to come back in play.
+             * -------+
+             *        | O
+             *        |    X
+             * -------+ O / -------
+             *           /
+             *          0
+             */
+            block_point =
+                block_point.value() + Vector(ROBOT_MAX_RADIUS_METERS * 2 + x_shift_meters,
+                                             -ROBOT_MAX_RADIUS_METERS * 2);
```

Comment:
isn't it possible that the adjusted point is inside the defense area?

Reply:
i don't think this is possible? could you explain how

Reply:
Yeah, the Y shift always pushes away from defence area